cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+	AiHubNameEnvVarName = "AZUREAI_HUB_NAME"
+	// AiProjectNameEnvVarName is the environment variable name for the Azure AI project name
+	AiProjectNameEnvVarName = "AZUREAI_PROJECT_NAME"
+	// AiEnvironmentEnvVarName is the environment variable name for the Azure AI environment name
+	AiEnvironmentEnvVarName = "AZUREAI_ENVIRONMENT_NAME"
+	// AiModelEnvVarName is the environment variable name for the Azure AI model name
+	AiModelEnvVarName = "AZUREAI_MODEL_NAME"
+	// AiEndpointEnvVarName is the environment variable name for the Azure AI endpoint name
+	AiEndpointEnvVarName = "AZUREAI_ENDPOINT_NAME"
+	// AiDeploymentEnvVarName is the environment variable name for the Azure AI deployment name
+	AiDeploymentEnvVarName = "AZUREAI_DEPLOYMENT_NAME"
+	// AiFlowEnvVarName is the environment variable name for the Azure AI flow name
+	AiFlowEnvVarName = "AZUREAI_FLOW_NAME"
+)
+
+type AiHelper interface {
+	// RequiredExternalTools returns the required external tools for the AiHelper
+	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	// Initialize initializes the AiHelper
+	Initialize(ctx context.Context) error
+	// EnsureWorkspace ensures that the workspace exists
+	EnsureWorkspace(ctx context.Context, scope *ai.Scope) error
+	// CreateEnvironmentVersion creates a new environment version
+	CreateEnvironmentVersion(
+		ctx context.Context,
+		scope *ai.Scope,
+		serviceConfig *ServiceConfig,
+		config *ai.ComponentConfig,
+	) (*armmachinelearning.EnvironmentVersion, error)
+	// CreateModelVersion creates a new model version
+	CreateModelVersion(
+		ctx context.Context,
+		scope *ai.Scope,
+		serviceConfig *ServiceConfig,
+		config *ai.ComponentConfig,
+	) (*armmachinelearning.ModelVersion, error)
+	// GetEndpoint retrieves an online endpoint
+	GetEndpoint(ctx context.Context, scope *ai.Scope, endpointName string) (*armmachinelearning.OnlineEndpoint, error)
+	// DeployToEndpoint deploys a new online deployment to an online endpoint
+	DeployToEndpoint(
+		ctx context.Context,
+		scope *ai.Scope,
+		serviceConfig *ServiceConfig,
+		endpointName string,
+		config *ai.EndpointDeploymentConfig,
+	) (*armmachinelearning.OnlineDeployment, error)
+	// DeletePreviousDeployments deletes all previous deployments of an online endpoint except the one with 100% traffic
+	DeletePreviousDeployments(ctx context.Context, scope *ai.Scope, endpointName string) error
+	// UpdateTraffic updates the traffic distribution of an online endpoint for the specified deployment
+	UpdateTraffic(
+		ctx context.Context,
+		scope *ai.Scope,
+		endpointName string,
+		deploymentName string,
+	) (*armmachinelearning.OnlineEndpoint, error)
+	// CreateFlow creates a new flow
+	CreateFlow(
+		ctx context.Context,
+		scope *ai.Scope,
+		serviceConfig *ServiceConfig,
+		config *ai.ComponentConfig,
+	) (*ai.Flow, error)
+}
+
+// aiHelper provides helper functions for interacting with Azure Machine Learning resources
+type aiHelper struct {
```

Comment:
why is this a `helper` and not a `manager`?  
Seems and looks like the managers we have for other operations/flows to me.

Reply:
This `aiHelper` component is really just a collection of functions that wrap either AI/ML SDK components or our python bridge.  It is not really orchestrating or managing anything in particular so decided to call it a helper similar to our container helper.

Reply:
Can we then just use exported functions w/o the struct?

Reply:
It requires too many dependencies and don't want to have caller pass them.

Reply:
referencing some feedback from the team about managers/helpers with common state as a way to reduce the arguments list from exported functions: https://github.com/Azure/azure-dev/pull/3774#discussion_r1576796792

Reply:
Thanks for the feedback.  I don't think that applies here since all the functions defined here are used in the same context and used together.